plugins/flytekit-kf-pytorch/flytekitplugins/kfpytorch/task.py

```diff
@@ -72,11 +138,42 @@ def __init__(self, task_config: PyTorch, task_function: Callable, **kwargs):
             task_function,
             task_type=self._PYTORCH_TASK_TYPE,
             **kwargs,
+        ) 
+    
+    def _convert_replica_spec(self, replica_config: Union[Master, Worker]) -> pytorch_task.DistributedPyTorchTrainingReplicaSpec:
+        resources = convert_resources_to_resource_model(requests=replica_config.requests, limits=replica_config.limits)
+        replicas = 1
+        # Master should always have 1 replica
+        if not isinstance(replica_config, Master):
+            replicas = replica_config.replicas
+        return pytorch_task.DistributedPyTorchTrainingReplicaSpec(
+            replicas=replicas,
+            image=replica_config.image,
+            resources=resources.to_flyte_idl() if resources else None,
+            restart_policy=replica_config.restart_policy.value if replica_config.restart_policy else None,
         )
-
+        
+    def _convert_run_policy(self, run_policy: RunPolicy) -> kubeflow_common.RunPolicy:
+        return kubeflow_common.RunPolicy(
+            clean_pod_policy=run_policy.clean_pod_policy.value if run_policy.clean_pod_policy else None,
+            ttl_seconds_after_finished=run_policy.ttl_seconds_after_finished,
+            active_deadline_seconds=run_policy.active_deadline_seconds,
+            backoff_limit=run_policy.active_deadline_seconds,
+        )
+        
     def get_custom(self, settings: SerializationSettings) -> Dict[str, Any]:
-        job = DistributedPyTorchTrainingTask(workers=self.task_config.num_workers)
-        return MessageToDict(job)
+        worker = self._convert_replica_spec(self.task_config.worker)
+        # support v0 config for backwards compatibility
+        if (self.task_config.num_workers):
+            worker.replicas = self.task_config.num_workers
```

Comment:
Only comment is this (and similar in other plugins) means that the `v0` configuration overrides the `v1` configuration right? So
```
task_config=TfJob(
    num_workers=4,
    workers=WorkerSpec{
        replicas=3,
    },
)
```
would result in the number of workers being 4 right? Is this correct? I'm very happy to defer this decision.

Also, very much butchered the syntax, but I'm sure the idea is understood.

Reply:
maybe we can raise an error if users pass both v0 and v1 config.

Reply:
v0 configuration overrides the v1 configuration. that's correct, so in that case it will be 4 workers. Do we want a different overriding logic?

Reply:
> maybe we can raise an error if users pass both v0 and v1 config.

I second that. We should simply disallow setting both configs.

Reply:
added throwing error if users pass both v0 and v1 config.